doctr/utils/metrics.py

```diff
@@ -488,3 +488,112 @@ def reset(self) -> None:
         self.caseless_matches = 0
         self.unidecode_matches = 0
         self.unicase_matches = 0
+
+
+class DetectionMetric:
+    r"""Implements an object detection metric.
+
+    The aggregated metrics are computed as follows:
+
+    .. math::
+        \forall (B, C) \in \mathcal{B}^N \times \mathcal{C}^N,
+        \forall (\hat{B}, \hat{C}) \in \mathcal{B}^M \times \mathcal{C}^M, \\
+        Recall(B, \hat{B}, C, \hat{C}) = \frac{1}{N} \sum\limits_{i=1}^N h_{B,C}(\hat{B}_i, \hat{C}_i) \\
+        Precision(B, \hat{B}, C, \hat{C}) = \frac{1}{M} \sum\limits_{i=1}^N h_{B,C}(\hat{B}_i, \hat{C}_i) \\
+        meanIoU(B, \hat{B}) = \frac{1}{M} \sum\limits_{i=1}^M \max\limits_{j \in [1, N]}  IoU(\hat{B}_i, B_j)
+
+    with the function :math:`IoU(x, y)` being the Intersection over Union between bounding boxes :math:`x` and
+    :math:`y`, and the function :math:`h_{B, C}` defined as:
+
+    .. math::
+        \forall (b, c) \in \mathcal{B} \times \mathcal{C},
+        h_{B,C}(b, c) = \left\{
+            \begin{array}{ll}
+                1 & \mbox{if } b\mbox{ has been assigned to a given }B_j\mbox{ with an } \\
+                & IoU \geq 0.5 \mbox{ and that for this assignment, } c = C_j\\
+                0 & \mbox{otherwise.}
+            \end{array}
+        \right.
+
+    where :math:`\mathcal{B}` is the set of possible bounding boxes,
+    :math:`\mathcal{C}` is the set of possible class indices,
+    :math:`N` (number of ground truths) and :math:`M` (number of predictions) are strictly positive integers.
+
+    Example::
+        >>> import numpy as np
+        >>> from doctr.utils import DetectionMetric
+        >>> metric = DetectionMetric(iou_thresh=0.5)
+        >>> metric.update(np.asarray([[0, 0, 100, 100]]), np.asarray([[0, 0, 70, 70], [110, 95, 200, 150]]),
+        np.zeros(1, dtype=np.int64), np.array([0, 1], dtype=np.int64))
+        >>> metric.summary()
+
+    Args:
+        iou_thresh: minimum IoU to consider a pair of prediction and ground truth as a match
+    """
+
+    def __init__(
+        self,
+        iou_thresh: float = 0.5,
+        rotated_bbox: bool = False,
+        mask_shape: Tuple[int, int] = (1024, 1024),
+    ) -> None:
+        self.iou_thresh = iou_thresh
+        self.rotated_bbox = rotated_bbox
+        self.mask_shape = mask_shape
+        self.reset()
+
+    def update(
+        self,
+        gt_boxes: np.ndarray,
+        pred_boxes: np.ndarray,
+        gt_labels: np.ndarray,
+        pred_labels: np.ndarray,
+    ) -> None:
+
+        if gt_boxes.shape[0] != gt_labels.shape[0] or pred_boxes.shape[0] != pred_labels.shape[0]:
+            raise AssertionError("there should be the same number of boxes and string both for the ground truth "
+                                 "and the predictions")
+
+        # Compute IoU
```

Comment:
We are using `self.tot_iou += float(iou_mat.max(axis=1).sum())` , however` self.tot_iou` is not defined above.